### PR TITLE
Refactor PaymentResource to include data block

### PR DESCRIPTION
**Pull request by dalebradley**
CPS-207

PaymentResource needed to be refactored to include a data block, so separate internal and external data when using the API.

Also, updated unit tests

### CODE REVIEW

File: service/payment_test.go [hunk 1]
```diff
@@ -281,7 +278,7 @@ func TestUnitGetPayment(t *testing.T) {
 	Convey("Invalid cost", t, func() {
 		mock := dao.NewMockDAO(mockCtrl)
 		mockPaymentService := createMockPaymentService(mock, cfg)
-		mock.EXPECT().GetPaymentResource(gomock.Any()).Return(&models.PaymentResource{Amount: "x", Links: models.Links{Resource: "http://dummy-resource"}}, nil)
+		mock.EXPECT().GetPaymentResource(gomock.Any()).Return(&models.PaymentResource{"1234", models.PaymentResourceData{Amount: "x", Links: models.Links{Resource: "http://dummy-resource"}}}, nil)
```
Comment (jphillips2121) [inline service/payment_test.go]: go vet is giving a warning here for "composite literal uses unkeyed fields"

If you replace `&models.PaymentResource{"1234", models.PaymentResourceData{Amount: "x", Links: models.Links{Resource: "http://dummy-resource"}}}, nil)` 

with 

`&models.PaymentResource{ID: "1234", Data: models.PaymentResourceData{Amount: "x", Links: models.Links{Resource: "http://dummy-resource"}}}, nil)`

It'll key the fields and also prevent the unit tests from breaking in the future if you we add another field

File: service/payment_test.go [hunk 2]
```diff
@@ -303,7 +300,7 @@ func TestUnitGetPayment(t *testing.T) {
 	Convey("Amount mismatch", t, func() {
 		mock := dao.NewMockDAO(mockCtrl)
 		mockPaymentService := createMockPaymentService(mock, cfg)
-		mock.EXPECT().GetPaymentResource(gomock.Any()).Return(&models.PaymentResource{Amount: "100", Links: models.Links{Resource: "http://dummy-resource"}}, nil)
+		mock.EXPECT().GetPaymentResource(gomock.Any()).Return(&models.PaymentResource{"1234", models.PaymentResourceData{Amount: "100", Links: models.Links{Resource: "http://dummy-resource"}}}, nil)
```
Comment (jphillips2121) [inline service/payment_test.go]: Same here

File: service/payment_test.go [hunk 3]
```diff
@@ -323,7 +320,7 @@ func TestUnitGetPayment(t *testing.T) {
 	Convey("Get Payment session - success - Single cost", t, func() {
 		mock := dao.NewMockDAO(mockCtrl)
 		mockPaymentService := createMockPaymentService(mock, cfg)
-		mock.EXPECT().GetPaymentResource(gomock.Any()).Return(&models.PaymentResource{Amount: "10", Links: models.Links{Resource: "http://dummy-resource"}}, nil)
+		mock.EXPECT().GetPaymentResource(gomock.Any()).Return(&models.PaymentResource{"1234", models.PaymentResourceData{Amount: "10", Links: models.Links{Resource: "http://dummy-resource"}}}, nil)
```
Comment (jphillips2121) [inline service/payment_test.go]: And here

File: service/payment_test.go [hunk 4]
```diff
@@ -343,7 +340,7 @@ func TestUnitGetPayment(t *testing.T) {
 	Convey("Get Payment session - success - Multiple costs", t, func() {
 		mock := dao.NewMockDAO(mockCtrl)
 		mockPaymentService := createMockPaymentService(mock, cfg)
-		mock.EXPECT().GetPaymentResource(gomock.Any()).Return(&models.PaymentResource{Amount: "23", Links: models.Links{Resource: "http://dummy-resource"}}, nil)
+		mock.EXPECT().GetPaymentResource(gomock.Any()).Return(&models.PaymentResource{"1234", models.PaymentResourceData{Amount: "23", Links: models.Links{Resource: "http://dummy-resource"}}}, nil)
```
Comment (jphillips2121) [inline service/payment_test.go]: And here

File: models/payment.go [hunk 5]
```diff
@@ -10,9 +10,13 @@ type IncomingPaymentResourceRequest struct {
 	State       string `json:"state"`
 }
 
-// PaymentResource is the payment details to be stored in the DB and returned in the response
```
Comment (emead) [inline models/payment.go]: Comment on exported struct should not be deleted

**Pull request merged**